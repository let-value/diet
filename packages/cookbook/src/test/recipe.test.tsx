import { test, expect } from "bun:test";

import {
	jsx,
	Recipe,
	Directions,
	Step,
	Ingredients,
	Ingredient,
	Preparation,
	Measurement,
} from "@/scheme";

export const test_recipe = (
	<Recipe name="test recipe" meal="breakfast,lunch,dinner" servings="4">
		<Ingredients>
			<Ingredient name="ingredient1" quantity="1g" category="category1" />
			<Ingredient category="category2">ingredient2 (raw)</Ingredient>
			<Ingredient key="ingredient3" quantity="20 slices" category="category3">
				Fresh ingredient 3
			</Ingredient>
			<Ingredient name="ingredient4" quantity="1tsp" category="category4" />
			<Ingredient category="category5" quantity="1">
				ingredient5
			</Ingredient>
		</Ingredients>
		<Preparation>
			<Step>
				take <Ingredient manipulation="boil">ingredient1</Ingredient> and boil
				it at <Measurement>100 celsius</Measurement> for{" "}
				<Measurement>10 minutes</Measurement>
			</Step>
			<Step>
				then take{" "}
				<Ingredient quantity="2kg" manipulation="fry">
					ingredient2
				</Ingredient>{" "}
				and fry in on a pan
			</Step>
			<Step duration="10 minutes">
				after that take{" "}
				<Ingredient key="ingredient3" manipulation="boil">
					ingredient 3
				</Ingredient>{" "}
				and boil it for <Measurement>10 minutes</Measurement>
			</Step>
		</Preparation>
		<Directions>
			<Step>step 1</Step>
			<Step>step 2</Step>
			<Step>step 3</Step>
		</Directions>
	</Recipe>
);

test("test_recipe", () => {
	expect(test_recipe).toMatchSnapshot();
});

export const exotic_recipe = (
	<Recipe name="Exotic Fruit Salad" meal="dessert" servings="6">
		<Ingredients>
			<Ingredient name="dragon fruit" quantity="2" category="fruit" />
			<Ingredient name="mango" quantity="1" category="fruit" />
			<Ingredient name="kiwi" quantity="3" category="fruit" />
			<Ingredient name="pomegranate seeds" quantity="1 cup" category="fruit" />
			<Ingredient name="fresh mint leaves" quantity="15 g" category="herb" />
			<Ingredient name="lime juice" quantity="2 tbsp" category="liquid" />
			<Ingredient
				name="honey"
				quantity="1 tbsp"
				category="sweetener"
				optional
			/>
		</Ingredients>
		<Preparation>
			<Step>
				Peel and cube the <Ingredient>dragon fruit</Ingredient> and{" "}
				<Ingredient>mango</Ingredient>. Slice the{" "}
				<Ingredient name="kiwi">kiwis</Ingredient>.
			</Step>
			<Step>
				In a large bowl, combine the cubed <Ingredient name="dragon fruit" />,{" "}
				<Ingredient name="mango" />, sliced <Ingredient name="kiwi" />, and{" "}
				<Ingredient>pomegranate seeds</Ingredient>.
			</Step>
			<Step>
				Gently toss with <Ingredient>lime juice</Ingredient> and{" "}
				<Ingredient optional>honey</Ingredient> (if using).
			</Step>
			<Step duration="5 minutes">
				Let the salad sit for <Measurement>5 minutes</Measurement> to blend the
				flavors.
			</Step>
			<Step>
				Garnish with{" "}
				<Ingredient name="fresh mint leaves">fresh mint leaves</Ingredient>{" "}
				before serving.
			</Step>
		</Preparation>
		<Directions>
			<Step>Prepare and combine fruits</Step>
			<Step>Toss with dressing</Step>
			<Step>Let sit and blend flavors</Step>
			<Step>Garnish and serve</Step>
		</Directions>
	</Recipe>
);

test("mediterraneanDelight", () => {
	expect(exotic_recipe).toMatchSnapshot();
});
