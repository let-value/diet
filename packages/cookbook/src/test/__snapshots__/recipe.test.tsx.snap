// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`test_recipe 1`] = `
Recipe {
  "children": [
    Ingredients {
      "children": [
        Ingredient {
          "category": [
            "category1",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "ingredient1",
          "optional": false,
          "quantity": _Unit {
            "dimension": {
              "MASS": 1,
            },
            "fixed": false,
            "type": "Unit",
            "unitList": [
              {
                "power": 1,
                "prefix": "",
                "unit": {
                  "basePrefix": "k",
                  "dimension": {
                    "MASS": 1,
                  },
                  "formatPrefixes": [
                    "n",
                    "u",
                    "m",
                    "",
                    "k",
                  ],
                  "name": "g",
                  "offset": 0,
                  "prefixGroup": {
                    "E": 1000000000000000000,
                    "G": 1000000000,
                    "M": 1000000,
                    "P": 1000000000000000,
                    "T": 1000000000000,
                    "Y": 1000000000000000000000000,
                    "Z": 1000000000000000000000,
                    "a": 0.000000000000000001,
                    "c": 0.01,
                    "d": 0.1,
                    "da": 10,
                    "f": 0.000000000000001,
                    "h": 100,
                    "k": 1000,
                    "m": 0.001,
                    "n": 0.000000001,
                    "p": 0.000000000001,
                    "u": 0.000001,
                    "y": 0.000000000000000000000001,
                    "z": 0.000000000000000000001,
                  },
                  "quantity": "MASS",
                  "value": 0.001,
                },
              },
            ],
            "value": 1,
          },
        },
        Ingredient {
          "category": [
            "category2",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "ingredient2 (raw)",
          "optional": false,
          "quantity": undefined,
        },
        Ingredient {
          "category": [
            "category3",
          ],
          "description": undefined,
          "key": "ingredient3",
          "manipulation": [],
          "name": "Fresh ingredient 3",
          "optional": false,
          "quantity": _Unit {
            "dimension": {},
            "fixed": false,
            "type": "Unit",
            "unitList": [
              {
                "power": 1,
                "prefix": "",
                "unit": {
                  "basePrefix": undefined,
                  "dimension": {},
                  "formatPrefixes": undefined,
                  "name": "slices",
                  "offset": 0,
                  "prefixGroup": {},
                  "value": 1,
                },
              },
            ],
            "value": 20,
          },
        },
        Ingredient {
          "category": [
            "category4",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "ingredient4",
          "optional": false,
          "quantity": _Unit {
            "dimension": {
              "LENGTH": 3,
            },
            "fixed": false,
            "type": "Unit",
            "unitList": [
              {
                "power": 1,
                "prefix": "",
                "unit": {
                  "basePrefix": undefined,
                  "dimension": {
                    "LENGTH": 3,
                  },
                  "formatPrefixes": undefined,
                  "name": "tsp",
                  "offset": 0,
                  "prefixGroup": {},
                  "value": 0.00000492892159375,
                },
              },
            ],
            "value": 1,
          },
        },
        Ingredient {
          "category": [
            "category5",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "ingredient5",
          "optional": false,
          "quantity": _Unit {
            "dimension": {},
            "fixed": false,
            "type": "Unit",
            "unitList": [],
            "value": 1,
          },
        },
      ],
    },
    Preparation {
      "children": [
        Step {
          "children": [
            "take ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [
                "boil",
              ],
              "name": "ingredient1",
              "optional": false,
              "quantity": undefined,
            },
            " and boil it at ",
            Measurement {
              "quantity": _Unit {
                "dimension": {
                  "TEMPERATURE": 1,
                },
                "fixed": false,
                "type": "Unit",
                "unitList": [
                  {
                    "power": 1,
                    "prefix": "",
                    "unit": {
                      "basePrefix": undefined,
                      "dimension": {
                        "TEMPERATURE": 1,
                      },
                      "formatPrefixes": undefined,
                      "name": "celsius",
                      "offset": 273.15,
                      "prefixGroup": {},
                      "value": 1,
                    },
                  },
                ],
                "value": 100,
              },
              "scale": undefined,
            },
            " for",
            " ",
            Measurement {
              "quantity": _Unit {
                "dimension": {
                  "TIME": 1,
                },
                "fixed": false,
                "type": "Unit",
                "unitList": [
                  {
                    "power": 1,
                    "prefix": "",
                    "unit": {
                      "basePrefix": undefined,
                      "dimension": {
                        "TIME": 1,
                      },
                      "formatPrefixes": undefined,
                      "name": "minutes",
                      "offset": 0,
                      "prefixGroup": {},
                      "value": 60,
                    },
                  },
                ],
                "value": 10,
              },
              "scale": undefined,
            },
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "then take",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [
                "fry",
              ],
              "name": "ingredient2",
              "optional": false,
              "quantity": _Unit {
                "dimension": {
                  "MASS": 1,
                },
                "fixed": false,
                "type": "Unit",
                "unitList": [
                  {
                    "power": 1,
                    "prefix": "k",
                    "unit": {
                      "basePrefix": "k",
                      "dimension": {
                        "MASS": 1,
                      },
                      "formatPrefixes": [
                        "n",
                        "u",
                        "m",
                        "",
                        "k",
                      ],
                      "name": "g",
                      "offset": 0,
                      "prefixGroup": {
                        "E": 1000000000000000000,
                        "G": 1000000000,
                        "M": 1000000,
                        "P": 1000000000000000,
                        "T": 1000000000000,
                        "Y": 1000000000000000000000000,
                        "Z": 1000000000000000000000,
                        "a": 0.000000000000000001,
                        "c": 0.01,
                        "d": 0.1,
                        "da": 10,
                        "f": 0.000000000000001,
                        "h": 100,
                        "k": 1000,
                        "m": 0.001,
                        "n": 0.000000001,
                        "p": 0.000000000001,
                        "u": 0.000001,
                        "y": 0.000000000000000000000001,
                        "z": 0.000000000000000000001,
                      },
                      "quantity": "MASS",
                      "value": 0.001,
                    },
                  },
                ],
                "value": 2,
              },
            },
            " ",
            "and fry in on a pan",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "after that take",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": "ingredient3",
              "manipulation": [
                "boil",
              ],
              "name": "ingredient 3",
              "optional": false,
              "quantity": undefined,
            },
            " ",
            "and boil it for ",
            Measurement {
              "quantity": _Unit {
                "dimension": {
                  "TIME": 1,
                },
                "fixed": false,
                "type": "Unit",
                "unitList": [
                  {
                    "power": 1,
                    "prefix": "",
                    "unit": {
                      "basePrefix": undefined,
                      "dimension": {
                        "TIME": 1,
                      },
                      "formatPrefixes": undefined,
                      "name": "minutes",
                      "offset": 0,
                      "prefixGroup": {},
                      "value": 60,
                    },
                  },
                ],
                "value": 10,
              },
              "scale": undefined,
            },
          ],
          "duration": _Unit {
            "dimension": {
              "TIME": 1,
            },
            "fixed": false,
            "type": "Unit",
            "unitList": [
              {
                "power": 1,
                "prefix": "",
                "unit": {
                  "basePrefix": undefined,
                  "dimension": {
                    "TIME": 1,
                  },
                  "formatPrefixes": undefined,
                  "name": "minutes",
                  "offset": 0,
                  "prefixGroup": {},
                  "value": 60,
                },
              },
            ],
            "value": 10,
          },
        },
      ],
    },
    Directions {
      "children": [
        Step {
          "children": [
            "step 1",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "step 2",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "step 3",
          ],
          "duration": undefined,
        },
      ],
    },
  ],
  "description": undefined,
  "meal": [
    "breakfast",
    "lunch",
    "dinner",
  ],
  "name": "test recipe",
  "servings": _Unit {
    "dimension": {},
    "fixed": false,
    "type": "Unit",
    "unitList": [],
    "value": 4,
  },
}
`;

exports[`mediterraneanDelight 1`] = `
Recipe {
  "children": [
    Ingredients {
      "children": [
        Ingredient {
          "category": [
            "fruit",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "dragon fruit",
          "optional": false,
          "quantity": _Unit {
            "dimension": {},
            "fixed": false,
            "type": "Unit",
            "unitList": [],
            "value": 2,
          },
        },
        Ingredient {
          "category": [
            "fruit",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "mango",
          "optional": false,
          "quantity": _Unit {
            "dimension": {},
            "fixed": false,
            "type": "Unit",
            "unitList": [],
            "value": 1,
          },
        },
        Ingredient {
          "category": [
            "fruit",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "kiwi",
          "optional": false,
          "quantity": _Unit {
            "dimension": {},
            "fixed": false,
            "type": "Unit",
            "unitList": [],
            "value": 3,
          },
        },
        Ingredient {
          "category": [
            "fruit",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "pomegranate seeds",
          "optional": false,
          "quantity": _Unit {
            "dimension": {
              "LENGTH": 3,
            },
            "fixed": false,
            "type": "Unit",
            "unitList": [
              {
                "power": 1,
                "prefix": "",
                "unit": {
                  "basePrefix": undefined,
                  "dimension": {
                    "LENGTH": 3,
                  },
                  "formatPrefixes": undefined,
                  "name": "cup",
                  "offset": 0,
                  "prefixGroup": {},
                  "value": 0.0002365882365,
                },
              },
            ],
            "value": 1,
          },
        },
        Ingredient {
          "category": [
            "herb",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "fresh mint leaves",
          "optional": false,
          "quantity": undefined,
        },
        Ingredient {
          "category": [
            "liquid",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "lime juice",
          "optional": false,
          "quantity": _Unit {
            "dimension": {
              "LENGTH": 3,
            },
            "fixed": false,
            "type": "Unit",
            "unitList": [
              {
                "power": 1,
                "prefix": "",
                "unit": {
                  "basePrefix": undefined,
                  "dimension": {
                    "LENGTH": 3,
                  },
                  "formatPrefixes": undefined,
                  "name": "tbsp",
                  "offset": 0,
                  "prefixGroup": {},
                  "value": 0.000014786764781249999,
                },
              },
            ],
            "value": 2,
          },
        },
        Ingredient {
          "category": [
            "sweetener",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "honey",
          "optional": true,
          "quantity": _Unit {
            "dimension": {
              "LENGTH": 3,
            },
            "fixed": false,
            "type": "Unit",
            "unitList": [
              {
                "power": 1,
                "prefix": "",
                "unit": {
                  "basePrefix": undefined,
                  "dimension": {
                    "LENGTH": 3,
                  },
                  "formatPrefixes": undefined,
                  "name": "tbsp",
                  "offset": 0,
                  "prefixGroup": {},
                  "value": 0.000014786764781249999,
                },
              },
            ],
            "value": 1,
          },
        },
      ],
    },
    Preparation {
      "children": [
        Step {
          "children": [
            "Peel and cube the ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "dragon fruit",
              "optional": false,
              "quantity": undefined,
            },
            " and",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "mango",
              "optional": false,
              "quantity": undefined,
            },
            ". Slice the",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "kiwi",
              "optional": false,
              "quantity": undefined,
            },
            ".",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "In a large bowl, combine the cubed ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "dragon fruit",
              "optional": false,
              "quantity": undefined,
            },
            ",",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "mango",
              "optional": false,
              "quantity": undefined,
            },
            ", sliced ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "kiwi",
              "optional": false,
              "quantity": undefined,
            },
            ", and",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "pomegranate seeds",
              "optional": false,
              "quantity": undefined,
            },
            ".",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "Gently toss with ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "lime juice",
              "optional": false,
              "quantity": undefined,
            },
            " and",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "honey",
              "optional": true,
              "quantity": undefined,
            },
            " (if using).",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "Let the salad sit for ",
            Measurement {
              "quantity": _Unit {
                "dimension": {
                  "TIME": 1,
                },
                "fixed": false,
                "type": "Unit",
                "unitList": [
                  {
                    "power": 1,
                    "prefix": "",
                    "unit": {
                      "basePrefix": undefined,
                      "dimension": {
                        "TIME": 1,
                      },
                      "formatPrefixes": undefined,
                      "name": "minutes",
                      "offset": 0,
                      "prefixGroup": {},
                      "value": 60,
                    },
                  },
                ],
                "value": 5,
              },
              "scale": undefined,
            },
            " to blend the flavors.",
          ],
          "duration": _Unit {
            "dimension": {
              "TIME": 1,
            },
            "fixed": false,
            "type": "Unit",
            "unitList": [
              {
                "power": 1,
                "prefix": "",
                "unit": {
                  "basePrefix": undefined,
                  "dimension": {
                    "TIME": 1,
                  },
                  "formatPrefixes": undefined,
                  "name": "minutes",
                  "offset": 0,
                  "prefixGroup": {},
                  "value": 60,
                },
              },
            ],
            "value": 5,
          },
        },
        Step {
          "children": [
            "Garnish with",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [],
              "name": "fresh mint leaves",
              "optional": false,
              "quantity": undefined,
            },
            " ",
            "before serving.",
          ],
          "duration": undefined,
        },
      ],
    },
    Directions {
      "children": [
        Step {
          "children": [
            "Prepare and combine fruits",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "Toss with dressing",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "Let sit and blend flavors",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "Garnish and serve",
          ],
          "duration": undefined,
        },
      ],
    },
  ],
  "description": undefined,
  "meal": [
    "dessert",
  ],
  "name": "Exotic Fruit Salad",
  "servings": _Unit {
    "dimension": {},
    "fixed": false,
    "type": "Unit",
    "unitList": [],
    "value": 6,
  },
}
`;
